src/contract-ui/tabs/edit-extensions/components/InstalledExtensionsTable.tsx

```diff
@@ -0,0 +1,347 @@
+            <TableHeading> Description </TableHeading>
+            <TableHeading> Publisher Address </TableHeading>
+            <TableHeading> Extension Address </TableHeading>
+            <TableHeading> Version </TableHeading>
+            {ownerAccount && <TableHeading> Remove </TableHeading>}
+          </Thead>
+
+          <Tbody>
+            {installedExtensions.isLoading ? (
+              <>
+                <SkeletonRow ownerAccount={ownerAccount} />
+                <SkeletonRow ownerAccount={ownerAccount} />
+                <SkeletonRow ownerAccount={ownerAccount} />
+              </>
+            ) : (
+              <>
+                {installedExtensions.data?.map((e, i) => (
+                  <ExtensionRow
+                    key={i}
+                    extensionAddress={e}
+                    contract={props.contract}
+                    onRemoveExtension={props.refetchExtensions}
+                    ownerAccount={ownerAccount}
+                  />
+                ))}
+              </>
+            )}
+          </Tbody>
+        </Table>
+      </TableContainer>
+    </>
+  );
+};
+
+function SkeletonRow(props: { ownerAccount?: Account }) {
+  return (
+    <Tr borderBottomWidth={1} _last={{ borderBottomWidth: 0 }}>
+      <TableData>
+        <Skeleton height="24px" rounded="xl" />
+      </TableData>
+      <TableData>
+        <Skeleton height="24px" rounded="xl" />
+      </TableData>
+      <TableData>
+        <Skeleton height="24px" rounded="xl" />
+      </TableData>
+      <TableData>
+        <Skeleton height="24px" rounded="xl" />
+      </TableData>
+
+      {/* Version */}
+      <TableData>
+        <Skeleton height="24px" rounded="xl" />
+      </TableData>
+
+      {/* Remove */}
+      {props.ownerAccount && (
+        <TableData>
+          <Skeleton height="24px" rounded="xl" />
+        </TableData>
+      )}
+    </Tr>
+  );
+}
+
+function ExtensionRow(props: {
+  extensionAddress: string;
+  contract: ContractOptions;
+  onRemoveExtension: () => void;
+  ownerAccount?: Account;
+}) {
+  const { contract, extensionAddress, ownerAccount } = props;
+  const toast = useToast();
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  const contractInfo = useExtensionContractInfo(props.extensionAddress);
+
+  const uninstallMutation = useMutation({
+    mutationFn: async (account: Account) => {
+      const uninstallTransaction = uninstallExtensionByProxy({
+        contract,
+        chain: contract.chain,
+        client: contract.client,
+        extensionProxyAddress: extensionAddress,
+        extensionData: "0x",
+      });
+
+      const txResult = await sendTransaction({
+        transaction: uninstallTransaction,
+        account,
+      });
+
+      await waitForReceipt(txResult);
+    },
+  });
+
+  const handleRemove = async () => {
+    if (!ownerAccount) {
+      return;
+    }
+
+    onClose();
+
+    try {
+      await uninstallMutation.mutateAsync(ownerAccount);
```

Comment:
same thing with using `mutate` instead and moving the async callbacks onto the query props